soda_machine.rb

```diff
@@ -7,9 +7,17 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.length
   end
 
   def find_soda(soda_brand)
+    # @sodas.select{|key, hash| hash["brand"] == soda_brand }
+    # @sodas.key({"brand" => soda_brand})
+    # @sodas.select {|e| e == soda_brand}
+    # @sodas.detect {|e| e == soda_brand}
+    @sodas.find {|soda| soda.brand == soda_brand}
```

Comment:
Nice job here, looks like you found the method that worked best for you in this situation.